libs/community/langchain_community/vectorstores/cratedb/base.py

```diff
@@ -0,0 +1,453 @@
+from __future__ import annotations
+
+import enum
+import math
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+)
+
+import sqlalchemy
+from langchain.docstore.document import Document
+from langchain.schema.embeddings import Embeddings
+from langchain.utils import get_from_dict_or_env
+from langchain.vectorstores.pgvector import PGVector
+from sqlalchemy.orm import sessionmaker
+
+from langchain_community.vectorstores.cratedb.model import ModelFactory
+
+
+class DistanceStrategy(str, enum.Enum):
+    """Enumerator of the Distance strategies."""
+
+    EUCLIDEAN = "euclidean"
+    COSINE = "cosine"
+    MAX_INNER_PRODUCT = "inner"
+
+
+DEFAULT_DISTANCE_STRATEGY = DistanceStrategy.EUCLIDEAN
+
+
+_LANGCHAIN_DEFAULT_COLLECTION_NAME = "langchain"
+
+
+def _results_to_docs(docs_and_scores: Any) -> List[Document]:
+    """Return docs from docs and scores."""
+    return [doc for doc, _ in docs_and_scores]
+
+
+class CrateDBVectorSearch(PGVector):
+    """`CrateDB` vector store.
+
+    To use it, you should have the ``crate[sqlalchemy]`` python package installed.
```

Comment:
Well, time flies. Please adjust package names.
```suggestion
    To use it, you should have the ``sqlalchemy-cratedb`` python package installed.
```

Reply:
Fixed per ffda5c8853f.